kernel/src/table_changes/physical_to_logical.rs

```diff
@@ -104,7 +106,16 @@ pub(crate) fn get_cdf_transform_expr(
     let cdf_values = get_cdf_columns(&state_info.logical_schema, scan_file)?;
     partition_values.extend(cdf_values);
 
-    get_transform_expr(transform_spec, partition_values, physical_schema)
+    let expr = get_transform_expr(transform_spec, partition_values, physical_schema)?;
```

Comment:
would it not make more sense to do the `is_identity` check inside `get_transform_expr` and have _that_ return: `DeltaResult<Option<ExpressionRef>>`?

Reply:
Yeah i think it'd look like this:
```rust
if transform_spec.empty() {
    return Ok(None);
}
// Regular get_transform_expr code
let mut transform = Transform::new_top_level();
```

that way, we avoid any allocation of expression.